tests/pkcli/ci_test.py

```diff
@@ -4,25 +4,24 @@
 :copyright: Copyright (c) 2022 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
+from pykern import pkunit
+from pykern.pkcli import ci
 from pykern.pkdebug import pkdp
-import re
-import pytest
 
 
-def test_run():
-    from pykern.pkcli import ci
-    from pykern import pkio
-    from pykern import pkunit
-
-    for d in pkunit.case_dirs("run"):
+def test_check_eof_newline():
```

Comment:
You have three cases which are the same. I think you could use the case_dir name to determine which method to call with a simple regex and renaming the directories. For example, rename run1.* to run-1.* and then you could even use str.split to get the name of the function.

Reply:
is this what you had in mind? a9e80a0beda32d4163def682c4cffe18637e6076

Reply:
Almost. _test could be test_all:

```py
def test_all():
    from pykern import pkunit
    from pykern.pkcli import ci

    for d in pkunit.case_dirs(case_dir):
        with pkunit.ExceptToFile():
            getattr(ci, d.basename.split("-")[0])()
```

Reply:
It's fully declarative. The directories describe the tests and the test script is just the interpreter, which will be just one function, does not need to be changed when new kinds of tests are added.

Reply:
The code you gave for test_all still uses case_dir. I don't see how this works without adding another loop around it like `for case_dir in ("check_eof_newline", "check_prints", "run")`. Could you explain that?

Reply:
Sorry I meant to remove case_dir as an argument to case_dirs. It's not necessary. If you read my original comment, it explains how this works. Try out the suggested file renaming.

Reply:
92d66abb3d1b0843ba74b3d09480199b674dd55b